Fix doc on new DefaultTrimmerAction behavior (#33618)
dotnet/docs: docs/core/compatibility/deployment/7.0/deprecated-trimmer-default-action.md

## Previous behavior

> Previously, only assemblies that were opted-in with `<IsTrimmable>true</IsTrimmable>` in the library project file were trimmed with the action specified by the `TrimmerDefaultAction`. The default value for that property was `copy` in .NET 6. While apps with warnings were more likely to work with this behavior, they still couldn't be guaranteed to not affect behavior. In addition, this caused apps to be larger than if the entire app was trimmed.

Reviewer: ```suggestion
Previously, only assemblies that were opted-in with `<IsTrimmable>true</IsTrimmable>` in the library project file were trimmed with the action specified by `TrimmerDefaultAction`. In .NET 6, the default value for that property was `copy`. While apps with warnings were more likely to work when this action was specified, run-time behavior could still be affected. In addition, the `copy` action caused apps to be larger than if the entire app was trimmed.
```

Reviewer: Trying to avoid the double negative.

Author: Couldn't commit this suggestion directly, but added it in a different commit.

## New behavior

> Starting in .NET 7, the property `TrimmerDefaultAction` is ignored and publishing behaves as if it was set to `link` all the time. This means all assemblies will be fully trimmed, whether they opt in or not. As a result, applications with trim warnings may see changes in behavior or runtime exceptions. See [All assemblies trimmed by default](./trim-all-assemblies.md) for more information and instructions for restoring the previous behavior.

Reviewer: ```suggestion
Starting in .NET 7, the property `TrimmerDefaultAction` is ignored and publishing behaves as if it was set to `link` all the time. This means all assemblies are fully trimmed, whether they opt in or not. As a result, applications with trim warnings may see changes in behavior or run-time exceptions. For more information and instrutions for restoring the previous behavior, see [All assemblies trimmed by default](./trim-all-assemblies.md).
```
